src/extlibs/hrvo/hrvo_agent.cpp

```diff
@@ -34,138 +34,199 @@
 
 #include <algorithm>
 #include <cmath>
-#include <iostream>
 #include <limits>
 
-#include "kd_tree.h"
 #include "path.h"
 #include "proto/message_translation/tbots_geometry.h"
+#include "software/geom/algorithms/intersection.h"
 #include "software/geom/vector.h"
 
-HRVOAgent::HRVOAgent(HRVOSimulator *simulator, const Vector &position, float neighborDist,
-                     std::size_t maxNeighbors, float radius, const Vector &velocity,
-                     float maxAccel, AgentPath &path, float prefSpeed, float maxSpeed,
-                     float uncertaintyOffset)
+HRVOAgent::HRVOAgent(HRVOSimulator *simulator, const Vector &position,
+                     float max_neighbor_dist, std::size_t maxNeighbors, float radius,
+                     const Vector &velocity, float maxAccel, AgentPath &path,
+                     float maxSpeed)
     : Agent(simulator, position, radius, velocity, velocity, maxSpeed, maxAccel, path),
       maxNeighbors_(maxNeighbors),
-      neighborDist_(neighborDist),
-      prefSpeed_(prefSpeed),
-      uncertaintyOffset_(uncertaintyOffset)
+      max_neighbor_dist(max_neighbor_dist),
+      prefSpeed_(max_speed_ * PREF_SPEED_SCALE),
+      // TODO (#2676): This config should be dependency injected and updated when a
+      // parameter is changed
+      obstacle_factory(TbotsProto::RobotNavigationObstacleConfig()),
+      ball_obstacle(std::nullopt)
 {
 }
 
-void HRVOAgent::computeNeighbors()
+void HRVOAgent::updatePrimitive(const TbotsProto::Primitive &new_primitive,
+                                const World &world)
+{
+    AgentPath path;
+    static_obstacles.clear();
+    ball_obstacle = std::nullopt;
+    if (new_primitive.has_move())
+    {
+        const auto &motion_control = new_primitive.move().motion_control();
+        float speed_at_dest        = new_primitive.move().final_speed_m_per_s();
+        float new_max_speed        = new_primitive.move().max_speed_m_per_s();
+        setMaxSpeed(new_max_speed);
+        setPreferredSpeed(new_max_speed * PREF_SPEED_SCALE);
+
+        // TODO (#2418): Update implementation of Primitive to support
+        // multiple path points and remove this check
+        CHECK(motion_control.path().points().size() >= 2)
+            << "Empty path: " << motion_control.path().points().size() << std::endl;
```

Comment:
We should handle this properly now, since the final speed for a multi-point path won't necessarily make sense for speeds at intermediate points.

Reply:
Discussed offline. In short, this currently not a high priority, as a result, it will be pushed back.